src/library/scala/collection/SeqView.scala

```diff
@@ -30,16 +30,42 @@ trait SeqView[+A] extends SeqOps[A, View, View[A]] with View[A] {
   def concat[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def appendedAll[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def prependedAll[B >: A](prefix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(prefix, this)
+
+  override def sorted[B >: A](implicit ord: Ordering[B]): SeqView[A] = {
+    val len = this.length // does not evaluate any view operations; checks `underlying.length`
+    new SeqView.Id(
+      if (len == 0) Nil
+      else if (len == 1) List(head)
+      else {
+        val arr = new Array[AnyRef](len)
+        var i = 0
+        for (x <- this) {
+          arr(i) = x.asInstanceOf[AnyRef]
+          i += 1
+        }
```

Comment:
I think it’d be even better with a while loop instead of a `foreach`.

Reply:
presumably this applies to the base impl as well?

Reply:
(this section vanishes in the new impl)